### ISSUE TITLE
importing complex objects from multiple (5+) JOINs - best practice?

### ISSUE THREAD

**Issue by Taskle** [jprante/elasticsearch-jdbc#351]
Hi,

TLDR version: what's the best practice to import complex structured objects from multiple tables where many JOINs (i.e. 5-10, not just 1 or 2) are involved with unrelated data needing to be indexed as properties of the same object?

Details...
Most examples/use cases I've seen regarding using JOINs to import structured/nested objects are for relatively simple MySQL queries in which a JOIN from one or two tables gets all the needed data. In our case, we have several 1-to-many/many-to-many tables that we'd like to denormalize into the ES index. I have a basic working example with 1 many-to-many relationship imported as nested objects, but I'm unclear from the documentation / heavy research into the code what method is best to proceed because I don't see any examples of several different tables being JOINed where the data in the JOINed tables is unrelated.

In our case, we are powering doctor search (millions of rows). Doctors can have multiple specialties, languages, treatments, insurances, etc. Each of these is its own many-to-many table we would need to JOIN. We'd like to denormalize all of this data into individual doctor objects with lists of the above data as subproperties. From what I've read, there's no way to use multiple queries to populate a single object; it all has to be done via a single query (I've tested this and confirmed that using the same _id just overwrites what was there previously). To join all of these tables in a single query for a river would require repeating a lot of the same data over and over, and I'm unclear if that's the right approach (and even if it's feasible given millions of rows of doctors).

What's the best practice to import complex structured objects from multiple tables where many JOINs are involved with unrelated data needing to be indexed as properties of the same object?

Thanks,
Dave
